src/system.c

```diff
@@ -327,10 +343,14 @@ int sys_filter_load(struct db_filter_col *col, bool rawrc)
 		if (col->attr.spec_allow)
 			flgs |= SECCOMP_FILTER_FLAG_SPEC_ALLOW;
 		rc = syscall(_nr_seccomp, SECCOMP_SET_MODE_FILTER, flgs, prgm);
-		if (rc > 0 && col->attr.tsync_enable)
+		if (tsync_notify && rc > 0) {
+			/* return 0 on NEW_LISTENER success, but save the fd */
+			col->notify_fd = rc;
+			rc = 0;
+		} else if (rc > 0 && col->attr.tsync_enable) {
 			/* always return -ESRCH if we fail to sync threads */
```

Comment:
nitpick - this comment is misleading now that we're setting errno.

Reply:
You lost me on the comment, but in looking at this again I realized that in addition to setting `errno = ESRCH` we also need to set `rc = -errno` so that we don't end up with a positive value in `rc`.